File: logging.rb
#require "logger"

class Logger

  ERROR = 3
  WARNING = 2
  INFO = 1
  DEBUG = 0

  attr_reader(:level)
  attr_writer(:level)

  def initialize(stream)
    @level = INFO
  end

  def error(msg)
    puts "ERROR: " + msg
  end

  def warn(msg)
    puts "WARN: " + msg
  end

  def debug(msg)
    puts msg
  end
end

$log = Logger.new(STDOUT)


# change $log.level to Logger::DEBUG, etc. as you need
$log.level=Logger::DEBUG

# change $debug to true to see all the debug logs
# NB: note this slows down everything if $debug is true even if the log level is not DEBUG
$debug = true
$debug_group = false

File: js/Logging.js
//Translated from logging.rb using babyruby2js
'use strict';

var main = require('./main');
//require 'logger';
main.log = new main.Logger(main.STDOUT);
// change $log.level to Logger::DEBUG, etc. as you need
main.log.level=(main.Logger.DEBUG);
// change $debug to true to see all the debug logs
// NB: note this slows down everything if $debug is true even if the log level is not DEBUG
main.debug = true;
main.debugGroup = false;
// W02: unknown class supposed to be attached to main: Logger
// W02: unknown constant supposed to be attached to main: STDOUT
// E02: unknown method: level=(...)
// W02: unknown class supposed to be attached to main: Logger
